pkg/services/ngalert/api/api_ruler.go

```diff
@@ -509,6 +515,20 @@ func validateQueries(ctx context.Context, groupChanges *store.GroupDelta, valida
 	return nil
 }
 
+// shouldValidate returns true if the rule is not paused and there are changes in the rule that are not ignored
+func shouldValidate(delta store.RuleDelta) bool {
+	diffs := make(map[string]bool)
+	for _, diff := range delta.Diff {
+		diffs[diff.Path] = true
+	}
+	for _, field := range ignoreFieldsForValidate {
+		delete(diffs, field)
+	}
+
+	// TODO: consider also checking if rule will be paused after the update
+	return len(diffs) > 0
+}
```

Comment:
As far as I understand, shouldValidate always returns `true` unless all delta.Diff are supposed to be ignored. Therefore, I think this can be simpler and I bit more straightforward

```suggestion
func shouldValidate(delta store.RuleDelta) bool {
	for _, diff := range delta.Diff {
		if !slices.Contains(ignoreFieldsForValidate, diff.Path) {
			return false
		}
	}
	return true
}
```

this will exit as soon as diff path is not ignored.

Reply:
I can't think of any real examples that would be past O(N+1) in the worst case.

Reply:
oh, dumb me... actually, hypothetical complexity is M*N, but in reality it's just O(2).